packages/stellar-axelar-std-derive/src/into_event.rs

```diff
@@ -25,7 +27,11 @@ pub fn into_event(input: &DeriveInput) -> proc_macro2::TokenStream {
                 #(, IntoVal::<_, Val>::into_val(&self.#data_field_idents, env))*
             ];
 
-            env.events().publish(topics, data);
+            if #has_datum {
+                env.events().publish(topics, data.get(0));
+            } else {
+                env.events().publish(topics, data);
+            }
```

Comment:
```suggestion
            let data = match fields.datum {
                Some((datum_field_ident, _)) => self.#datum_field_ident,
                None => {
                    let (data_field_idents, data_types) = fields.data;
                    ...
                    data
                }
            }

            env.events().publish(topics, data);
```
Add `fields.datum` as a `Option<Val>` so it can be accessed more cleanly?